plugins/source/hubspot/resources/plugin/client.go

```diff
@@ -0,0 +1,156 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/cloudquery/cloudquery/plugins/source/hubspot/client"
+	"github.com/cloudquery/cloudquery/plugins/source/hubspot/client/spec"
+	"github.com/cloudquery/cloudquery/plugins/source/hubspot/resources/services/crm"
+	"github.com/cloudquery/plugin-sdk/v4/caser"
+	"github.com/cloudquery/plugin-sdk/v4/docs"
+	"github.com/cloudquery/plugin-sdk/v4/message"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"github.com/cloudquery/plugin-sdk/v4/scheduler"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/state"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/rs/zerolog"
+	"golang.org/x/exp/maps"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	maxMsgSize = 100 * 1024 * 1024 // 100 MiB
+)
+
+var customExceptions = map[string]string{
+	"crm":     "CRM",
+	"hubspot": "HubSpot",
+}
+
+type Client struct {
+	logger      zerolog.Logger
+	config      spec.Spec
+	backendConn *grpc.ClientConn
+	scheduler   *scheduler.Scheduler
+	options     plugin.NewClientOptions
+	allTables   schema.Tables
+	plugin.UnimplementedDestination
+}
+
+func (*Client) Close(_ context.Context) error {
+	return nil
+}
+
+func (c *Client) Tables(_ context.Context, options plugin.TableOptions) (schema.Tables, error) {
+	return c.allTables.FilterDfs(options.Tables, options.SkipTables, options.SkipDependentTables)
+}
+
+func (c *Client) Sync(ctx context.Context, options plugin.SyncOptions, res chan<- message.SyncMessage) error {
+	if c.options.NoConnection {
+		return fmt.Errorf("no connection")
+	}
+
+	tables, err := c.allTables.FilterDfs(options.Tables, options.SkipTables, options.SkipDependentTables)
+	if err != nil {
+		return err
+	}
+
+	var stateClient state.Client
+	if options.BackendOptions == nil {
+		c.logger.Info().Msg("No backend options provided, using no state backend")
+		stateClient = &state.NoOpClient{}
+		c.backendConn = nil
+	} else {
+		c.backendConn, err = grpc.DialContext(ctx, options.BackendOptions.Connection,
+			grpc.WithTransportCredentials(insecure.NewCredentials()),
+			grpc.WithDefaultCallOptions(
+				grpc.MaxCallRecvMsgSize(maxMsgSize),
+				grpc.MaxCallSendMsgSize(maxMsgSize),
+			),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to dial grpc source plugin at %s: %w", options.BackendOptions.Connection, err)
+		}
+		stateClient, err = state.NewClient(ctx, c.backendConn, options.BackendOptions.TableName)
```

Comment:
It seems that having `backendConn` as `Client` member isn't actually required. Let's get rid of it & only use it in this file.